core/store/src/lib.rs

```diff
@@ -300,10 +305,22 @@ pub fn create_store(path: &Path) -> Store {
     Store::new(db)
 }
 
-/// Creates a store which is unable to modify an existing RocksDB instance.
-/// Panics if a write operation is attempted.
-pub fn open_read_only_store(path: &Path) -> Store {
-    let db = Arc::new(RocksDB::new_read_only(path).expect("Failed to open the database"));
+#[derive(Default, Debug)]
+pub struct StoreConfig {
+    pub read_only: bool,
+    pub enable_statistics: bool,
```

Comment:
mention if this has any impact on performance

Reply:
Done